pilot/pkg/networking/core/v1alpha3/envoyfilter.go

```diff
@@ -212,6 +212,29 @@ func listenerMatch(in *plugin.InputParams, listenerIP net.IP,
 	return true
 }
 
+// Delete filter before insert, if envoyFilter has already existed in httpFilters
+func updateHTTPFilter(listenerName string, filterChain *listener.FilterChain, hcm *http_conn.HttpConnectionManager,
+	envoyFilter *networking.EnvoyFilter_Filter, isXDSMarshalingToAnyEnabled bool) {
+	deleteFilter := func(index int) {
+		hcm.HttpFilters = append(hcm.HttpFilters[:index], hcm.HttpFilters[index+1:]...)
+	}
+
+	hasFilter := func(name string) (exists bool, index int) {
```

Comment:
I think you can not check by filter name and delete it - It may not be true always. I think there can be two `envoy.lua` filters with different lua scripts in the filter list. We can not remove one of them.

Reply:
ok, and what if check by filter name and config content ?